File: C++/1035-password.cc
#include <iostream>
#include <vector>
#include <string>
#include <utility>


int main() {

    int n_user;
    std::cin >> n_user;

    std::vector<std::pair<std::string, std::string>> modified;
    for (int i=0; i<n_user; ++i) {
        std::string name, password;
        bool is_modified = false;
        std::cin >> name >> password;
        for (auto &c : password) {
            switch (c) {
                case '1': c = '@'; is_modified = true; break;
                case '0': c = '%'; is_modified = true; break;
                case 'l': c = 'L'; is_modified = true; break;
                case 'O': c = 'o'; is_modified = true; break;
                default: break;
            }
        }
        if (is_modified) {
            modified.push_back(std::make_pair(name, password));
        }
    }

    if (modified.size()) {
        std::cout << modified.size() << '\n';
        for (auto &name_passwd : modified) {
            std::cout << name_passwd.first << ' '
                      << name_passwd.second << '\n';
        }
    } else {
        std::cout << "There ";
        if (n_user == 1) {
            std::cout << "is 1 account";
        } else {
            std::cout << "are " << n_user << " accounts";
        }
        std::cout << " and no account is modified";
    }

    return 0;
}

File: Python/1035-password.py


REPLACE_DICT = {
    '1': '@',
    '0': '%',
    'l': 'L',
    'O': 'o'
}


def modify_passwd(passwd):
    new_passwd = []
    is_modified = False
    for c in passwd:
        try:
            new_passwd.append(REPLACE_DICT[c])
            is_modified = True
        except KeyError:
            new_passwd.append(c)
    if is_modified:
        return ''.join(new_passwd)
    else:
        return ''


if __name__ == '__main__':
    n_user = int(input())
    modified = []
    for _ in range(n_user):
        name, passwd = input().split()
        new_passwd = modify_passwd(passwd)
        if new_passwd:
            modified.append((name, new_passwd))
    if modified:
        print(len(modified))
        for name, passwd in modified:
            print(name, passwd)
    else:
        if n_user > 1:
            account = f'are {n_user} accounts'
        else:
            account = 'is 1 account'
        print(f'There {account} and no account is modified')
